lib/URI.pm

```diff
@@ -102,7 +102,7 @@ sub _fix_uric_escape_for_host_part {
   return if $_[0] !~ /%/;
 
   if ($_[0] =~ m,^((?:$URI::scheme_re:)?)//([^/?\#]*)(.*)$,os) {
-    my $orig          = $2;
+    my $orig          = $2 || return; #-- no authority part to unescape (issue#102)
```

Comment:
It's not clear to me why there is a `--` after the comment marker. Maybe this comment could be clearer.

> Return early if there is no authority part to unescape. Fixes #102.